version: "3.4"

services:
  vaultwarden:
    image: vaultwarden/server:latest
    networks:
      - network_cluster
    volumes:
      - homenas_vaultwarden:/data
    environment:
      - SIGNUPS_ALLOWED=false
      - INVITATIONS_ALLOWED=false
      - ADMIN_TOKEN=${VAULTWARDEN_TOKEN}
    restart: always
    deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.vaultwarden.rule=Host(`vaultwarden.cloudvalley.name`)"
        - "traefik.http.routers.vaultwarden.entrypoints=websecure"
        - "traefik.http.routers.vaultwarden.service=vaultwarden"
        - "traefik.http.routers.vaultwarden.middlewares=noauth-chain@file"
        - "traefik.http.services.vaultwarden.loadbalancer.server.port=80"     
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.labels.nas == primary
          
networks:
  network_cluster:
    external: true
     
x-common-keys-volume: &common-keys-volume
  type: nfs
  o: addr=192.168.11.99,rw,nfsvers=4
  
volumes:
  homenas_vaultwarden: 
    driver: local
    driver_opts:
      <<: *common-keys-volume
      device: :/volume1/docker/swarm/storage/vaultwarden/data